coderd/agentapi/subagent.go

```diff
@@ -71,12 +81,112 @@ func (a *SubAgentAPI) CreateSubAgent(ctx context.Context, req *agentproto.Create
 		return nil, xerrors.Errorf("insert sub agent: %w", err)
 	}
 
+	var appCreationErrors []*agentproto.CreateSubAgentResponse_AppCreationError
+	appSlugs := make(map[string]struct{})
+
+	for i, app := range req.Apps {
+		err := func() error {
+			slug := app.Slug
+			if slug == "" {
+				return codersdk.ValidationError{
+					Field:  "slug",
+					Detail: "must not be empty",
+				}
+			}
+			if !provisioner.AppSlugRegex.MatchString(slug) {
+				return codersdk.ValidationError{
+					Field:  "slug",
+					Detail: fmt.Sprintf("%q does not match regex %q", slug, provisioner.AppSlugRegex),
+				}
+			}
+			if _, exists := appSlugs[slug]; exists {
+				return codersdk.ValidationError{
+					Field:  "slug",
+					Detail: fmt.Sprintf("%q is already in use", slug),
+				}
+			}
+			appSlugs[slug] = struct{}{}
+
+			health := database.WorkspaceAppHealthDisabled
+			if app.Healthcheck == nil {
+				app.Healthcheck = &agentproto.CreateSubAgentRequest_App_Healthcheck{}
+			}
+			if app.Healthcheck.Url != "" {
+				health = database.WorkspaceAppHealthInitializing
+			}
+
+			sharingLevel := database.AppSharingLevelOwner
+			switch app.GetShare() {
+			case agentproto.CreateSubAgentRequest_App_AUTHENTICATED:
+				sharingLevel = database.AppSharingLevelAuthenticated
+			case agentproto.CreateSubAgentRequest_App_PUBLIC:
+				sharingLevel = database.AppSharingLevelPublic
+			}
+
+			openIn := database.WorkspaceAppOpenInSlimWindow
+			if app.GetOpenIn() == agentproto.CreateSubAgentRequest_App_TAB {
+				openIn = database.WorkspaceAppOpenInTab
+			}
```

Comment:
Perhaps we should skip the default and have `switch` for each type with a default that returns an error? Thinking about future changes to the API and potential silent breakages/incorrect behavior that will go unnoticed otherwise.